docker/Dockerfile-fargo-master

```diff
@@ -1,9 +1,9 @@
-FROM golang:1.6
-
-ENV GOROOT=/usr/local/go
+FROM golang:1.16
```

Comment:
Does this work for you? 
When I tested this Docker file against current code (by temporary adding
```
RUN git remote add sagikazarmark https://github.com/sagikazarmark/fargo.git
RUN git fetch sagikazarmark
RUN git checkout --track sagikazarmark/updates
```
)

It doesn't work. I'm not sure if new dependencies are not introducing some problems... If you could take a look at this please it would be great.

Reply:
To be more precise I run tests inside this container while running Eureka in sidecar containers along with this one (as per README.md)

Reply:
Yep, works for me. Tests are passing as well:

```dockerfile
FROM golang:1.16

RUN go get github.com/tools/godep

RUN git clone https://github.com/hudl/fargo.git /usr/local/src/fargo

WORKDIR /usr/local/src/fargo

RUN git remote add sagikazarmark https://github.com/sagikazarmark/fargo.git
RUN git fetch sagikazarmark
RUN git checkout --track sagikazarmark/updates

RUN go mod download

ENV SKIP_NET_TESTS=true SKIP_MARSHAL_TESTS=true
RUN go test -v ./...
```

Reply:
What's the exact error you see?

Reply:
With skipping net and marshal tests it works. I still would like those to be passing when eureka containers are running though.

Example error:
```
 * C:/hudl/go/src/github.com/hudl/fargo/tests/net_test.go
  Line 122:
  Expected: nil
  Actual:   'json: cannot unmarshal array into Go struct field GetAppResponseJson.application of type struct { *fargo.instance; Port fargo.inboundJSONFormatPort "json:\"port\""; SecurePort fargo.inboundJSONFormatPort "json:\"securePort\"" }'
  goroutine 9 [running]:
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/reporting/reports.go:143 +0x74
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/reporting/reports.go:103 +0x73
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/context.go:176 +0xe6
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/doc.go:125 +0x6a
  github.com/hudl/fargo/tests_test.TestGetApps.func2()
        C:/hudl/go/src/github.com/hudl/fargo/tests/net_test.go:122 +0x94
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/discovery.go:80 +0x2b
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/context.go:261 +0x129
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/context.go:110 +0xf5
  github.com/jtolds/gls.(*ContextManager).SetValues.func1(0x0)
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/context.go:97 +0x470
  github.com/jtolds/gls.EnsureGoroutineId.func1()
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/gid.go:24 +0x35
  github.com/jtolds/gls._m(0x0, 0xc00009d1e8)
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/stack_tags.go:108 +0x38
  github.com/jtolds/gls.github_com_jtolds_gls_markS(0x0, 0xc00009d1e8)
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/stack_tags.go:56 +0x3c
  github.com/jtolds/gls.addStackTag(...)
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/stack_tags.go:49
  github.com/jtolds/gls.EnsureGoroutineId(0xc0001a0a80)
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/gid.go:24 +0xfd
  github.com/jtolds/gls.(*ContextManager).SetValues(0xc000089e40, 0xc0001a0a20, 0xc0004c5040)
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/context.go:63 +0x159
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/context.go:105 +0x23c
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/doc.go:75 +0xc6
  github.com/hudl/fargo/tests_test.TestGetApps(0xc00049c180)
        C:/hudl/go/src/github.com/hudl/fargo/tests/net_test.go:120 +0x205
  testing.tRunner(0xc00049c180, 0xdb5238)
        C:/Program Files/Go/src/testing/testing.go:1193 +0xef
  created by testing.(*T).Run
        C:/Program Files/Go/src/testing/testing.go:1238 +0x2b3
```

Or

```

  * C:/hudl/go/src/github.com/hudl/fargo/tests/net_test.go
  Line 170:
  Expected collection to have length equal to [1], but it's length was [0] instead! contents: []
  goroutine 53 [running]:
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/reporting/reports.go:143 +0x74
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/reporting/reports.go:103 +0x73
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/context.go:176 +0xe6
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/doc.go:125 +0x6a
  github.com/hudl/fargo/tests_test.TestGetSingleInstanceByVIPAddress.func1(0xc0003cb400)
        C:/hudl/go/src/github.com/hudl/fargo/tests/net_test.go:170 +0x1a5
  github.com/hudl/fargo/tests_test.withCustomRegisteredInstance.func1()
        C:/hudl/go/src/github.com/hudl/fargo/tests/net_test.go:80 +0x2e2
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/discovery.go:80 +0x2b
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/context.go:261 +0x129
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/context.go:110 +0xf5
  github.com/jtolds/gls.(*ContextManager).SetValues.func1(0x0)
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/context.go:97 +0x470
  github.com/jtolds/gls.EnsureGoroutineId.func1()
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/gid.go:24 +0x35
  github.com/jtolds/gls._m(0x0, 0xc000334d50)
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/stack_tags.go:108 +0x38
  github.com/jtolds/gls.github_com_jtolds_gls_markS(0x0, 0xc000334d50)
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/stack_tags.go:56 +0x3c
  github.com/jtolds/gls.addStackTag(...)
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/stack_tags.go:49
  github.com/jtolds/gls.EnsureGoroutineId(0xc0002c6cf0)
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/gid.go:24 +0xfd
  github.com/jtolds/gls.(*ContextManager).SetValues(0xc000089e40, 0xc0002c6c90, 0xc00037eb60)
        C:/hudl/go/pkg/mod/github.com/jtolds/gls@v4.20.0+incompatible/context.go:63 +0x159
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/context.go:105 +0x23c
        C:/hudl/go/pkg/mod/github.com/smartystreets/goconvey@v1.6.4/convey/doc.go:75 +0xc6
  github.com/hudl/fargo/tests_test.TestGetSingleInstanceByVIPAddress(0xc000283500)
        C:/hudl/go/src/github.com/hudl/fargo/tests/net_test.go:166 +0x3e5
  testing.tRunner(0xc000283500, 0xdb5270)
        C:/Program Files/Go/src/testing/testing.go:1193 +0xef
  created by testing.(*T).Run
        C:/Program Files/Go/src/testing/testing.go:1238 +0x2b3

```